model_data_preprocess/postprocessing_E3SM_data_for_TC_analysis.sh

```diff
@@ -1,100 +1,84 @@
-# tempestremap and tempestextremes are built in e3sm-unified from version 1.5.0. This script has been tested using Cori login node.
-# To request an interactive session as follows: salloc --nodes=1 --partition=debug --time=30:00 -C haswell
+# tempestremap and tempestextremes are built in e3sm-unified from version 1.5.0. This script has been tested using Perlmutter login node.
+# To request an interactive session as follows: salloc --nodes=1 --partition=debug --time=30:00 -C cpu
 
-source /global/common/software/e3sm/anaconda_envs/load_latest_e3sm_unified_cori-haswell.sh
+source /global/common/software/e3sm/anaconda_envs/load_latest_e3sm_unified_pm-cpu.sh
 
 #For typical EAM v2 ne gp2 grids.
-start="0051"
-end="0060"
-caseid="20210528.v2rc3e.piControl.ne30pg2_EC30to60E2r2.chrysalis"
-drc_in=/global/cfs/cdirs/e3smdata/zppy_complete_run_nersc_output/${caseid}/archive/atm/hist
-# Warning: because somehow tempest-remap only can writes grid file on SCRATCH space. The resulted files will be moved to another path at the end.
-result_dir_fin=/global/cfs/cdirs/e3sm/e3sm_diags/postprocessed_e3sm_v2_data_for_e3sm_diags/${caseid}/
-mkdir -p $result_dir_fin
-result_dir=$SCRATCH/tests/tc-analysis/
-
+start="2000"
+end="2014"
+caseid="v2.LR.amip_0101"
+drc_in=/global/cfs/cdirs/e3smpub/E3SM_simulations/${caseid}/archive/atm/hist
+result_dir=/global/homes/c/chengzhu/tests/tc_analysis/tc-v2.LR.amip.${start}_${end}/
 
 #res = 30/120 for ne30/ne120 grids
 res=30               # res = 30/120 for ne30/ne120 grids
 pg2=true             # Set false for v1 production simulations
 atm_name="eam"       # Use "cam" for v1 production simulations
 
 
-##For typical v1 ne SE grids.
-#start="1880"
-#end="1881"
-#caseid="20180215.DECKv1b_H1.ne30_oEC.edison"
-#drc_in=/global/cfs/cdirs/e3smpub/E3SM_simulations/${caseid}/archive/atm/hist
-#result_dir=/global/cfs/cdirs/e3sm/e3sm_diags/postprocessed_e3sm_v1_data_for_e3sm_diags/${caseid}/tc-analysis/
-#
-##res = 30/120 for ne30/ne120 grids
-#res=30               # res = 30/120 for ne30/ne120 grids
-#pg2=false             # Set false for v1 production simulations
-#atm_name="cam"       # Use "cam" for v1 production simulations
-
-
-
 mkdir -p $result_dir
 file_name=${caseid}_${start}_${end}
 
 # Generate mesh files (.g).
-#/global/homes/p/paullric/tempestremap/bin/GenerateCSMesh --res $res --alt --file ${result_dir}outCSne$res.g
-GenerateCSMesh --res $res --alt --file ${result_dir}outCSne$res.g
-out_type="CGLL"
-# For v2 production simulation with pg2 grids:
 if $pg2; then
-    GenerateVolumetricMesh --in ${result_dir}outCSne$res.g  --out ${result_dir}outCSne$res.g --np 2 --uniform
+    # For v2 and v3 production simulation with pg2 grids:
+    GenerateCSMesh --res $res --alt --file ${result_dir}outCSMeshne$res.g
+    GenerateVolumetricMesh --in ${result_dir}outCSMeshne$res.g  --out ${result_dir}outCSne$res.g --np 2 --uniform
```

Comment:
`--in` and `--out` file names were the same before, which I suspect caused the intermittent issue when mesh file can't be written on `cfs`.